src/Microsoft.AspNet.Razor/RazorResources.resx

```diff
@@ -413,4 +413,10 @@ Instead, wrap the contents of the block in "{{}}":
   <data name="SourceLocationFilePathDoesNotMatch" xml:space="preserve">
     <value>Cannot perform '{1}' operations on '{0}' instances with different file paths.</value>
   </data>
+  <data name="TagHelperParseTreeRewriter_InconsistentTagStructure" xml:space="preserve">
+    <value>Tag helpers '{0}' and '{1}' targeting element '{2}' must not expect different tag structures. '{0}' expects tag structure '{3}' and '{1}' expects tag structure '{4}'.</value>
+  </data>
+  <data name="TagHelperParseTreeRewriter_EndTagTagHelperMustNotHaveAnEndTag" xml:space="preserve">
+    <value>Found an end '{0}' tag for tag helper '{1}' with tag structure '{2}'. Cannot parse end tags for '{2}' tag structure tag helpers.</value>
```

Comment:
First sentence looks sufficient for both of these messages. Suggest
```
Tag helpers '{0}' and '{1}' targeting element '{2}' must not expect different {3} values.
```
where caller passes in "TagStructure" because `nameof` won't work here -- the attribute isn't accessible.

And
```
Found an end tag (&lt;/{0}&gt;) for tag helper '{1}' with tag structure that disallows an end tag ('{2}').
```